tests/conftest.py

```diff
@@ -32,6 +33,39 @@
 collect_ignore = ["setup.py"]
 
 
+@pytest.fixture(autouse=True)
+def virtual_memory(mocker):
+    class VirtualMemory(NamedTuple):
+        total: int
+        available: int
+        percent: int
+        used: int
+        free: int
+
+    return mocker.patch(
+        "psutil.virtual_memory",
+        return_value=VirtualMemory(
+            total=1073741824,
+            available=5368709120,
+            # prevent dumping of smaller batches to db in process_udf_outputs
+            # we want to avoid this due to tests running concurrently (xdist)
+            percent=50,
+            used=5368709120,
+            free=5368709120,
+        ),
+    )
+
+
+@pytest.fixture(autouse=True)
+def per_thread_im_mem_db(mocker, worker_id):
+    if worker_id == "master":
+        return
+    mocker.patch(
+        "datachain.data_storage.sqlite._get_in_memory_uri",
+        return_value=f"file:in-mem-db-{worker_id}?mode=memory&cache=shared",
```

Comment:
from https://www.sqlite.org/draft/inmemorydb.html: 

If two or more distinct but shareable in-memory databases are needed in a single process, then the [mode=memory](https://www.sqlite.org/draft/uri.html#coreqp) query parameter can be used with a [URI filename](https://www.sqlite.org/draft/uri.html) to create a named in-memory database:

rc = sqlite3_open("file:memdb1?mode=memory&cache=shared", &db);

We need one per worker to stop them from running into one another.